requre/online_replacing.py

```diff
@@ -81,3 +87,188 @@ def _internal(*args, **kwargs):
         return _internal
 
     return decorator_cover
+
+
+def replace_module_match(
+    what: str,
+    decorate: Optional[Union[List[Callable], Callable]] = None,
+    replace: Optional[Callable] = None,
+    storage_file: Optional[str] = None,
+    storage_keys_strategy=StorageKeysInspectSimple,
+):
+    """
+    Decorator what helps you to replace/decorate functions/methods inside any already
+    imported module. It uses what as identifier what you want to replace, then you can
+    define if it will be decorated or replaced by given function.
+
+    Example usage to decorate request method of requests module:
+    @replace_module_match(what="requests.sessions.Session.request",
+                          decorate=RequestResponseHandling.decorator(
+                          item_list=["method", "url", "data"],
+                          map_item_list={"url": remove_password_from_url})
+
+    :param what: str - full path of function inside module
+    :param decorate: function decorator what will be applied to what, could be also list of
+                     decorators, to be able to apply more decorators on one function
+                     eg. store files and store output
+    :param replace: replace original function by given one
+    :param storage_file: path for storage file if you don't want to use default location
+    :param storage_keys_strategy: you can change key strategy for storing data
+                                  default simple one avoid to store stack information
+    """
+    if (decorate is None and replace is None) or (
+        decorate is not None and replace is not None
+    ):
+        raise ValueError("right one from [decorate, replace] parameter has to be set.")
+
+    def decorator_cover(func):
+        @functools.wraps(func)
+        def _internal(*args, **kwargs):
+            logger.info(
+                f"\n++++++ SEARCH {what} decorator={decorate} replace={replace}"
+            )
+            original_key_strategy = DataMiner().key_stategy_cls
+            # use simple strategy, to not store stack info as keys
+            DataMiner().key_stategy_cls = storage_keys_strategy
+            if storage_file:
+                PersistentObjectStorage().storage_file = storage_file
+            else:
+
+                if len(args):
+                    try:
+                        PersistentObjectStorage().storage_file = get_datafile_filename(
+                            args[0]
+                        )
+                    except NameError:
+                        PersistentObjectStorage().storage_file = get_datafile_filename(
+                            func
+                        )
+                else:
+                    PersistentObjectStorage().storage_file = get_datafile_filename(func)
+            os.makedirs(
+                os.path.dirname(PersistentObjectStorage().storage_file), exist_ok=True
+            )
+            original_storage_file = PersistentObjectStorage().storage_file
```

Comment:
It would be nice to have a separate helper function for each logical unit of the workflow.

e.g. here, exclude this to some function managing the storage(_file).

Reply:
Sorry for the duplicate comment. You made some changes during my review.